Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1219,6 +1219,35 @@ static NavigationNavigationType determineNavigationType(FrameLoadType loadType,
     return NavigationNavigationType::Push;
 }
 
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#url-and-history-update-steps
+void FrameLoader::updateURLAndHistory(const URL& newURL, RefPtr<SerializedScriptValue>&& stateObject, NavigationHistoryBehavior historyHandling)
+{
+    RefPtr document = m_frame->document();
+    RefPtr documentLoader = protectedDocumentLoader();
```

Comment:
This refs twice I think, common is to do:
RefPtr documentLoader = documentLoader();